Sources/Workspace/Workspace.swift

```diff
@@ -516,6 +516,18 @@ public class Workspace {
             RepositoryPackageConstraint(
                 container: dependency.repository, requirement: requirement))
 
+        // Add unversioned constraint for edited packages.
+        let currentManifests = loadDependencyManifests(rootManifests: rootManifests, engine: engine)
+        for (externalManifest, managedDependency) in currentManifests.dependencies {
+            switch managedDependency.state {
+            case .checkout: continue
+            case .unmanaged, .edited: break
+            }
+            let specifier = RepositorySpecifier(url: externalManifest.url)
+            let dependencies = externalManifest.package.dependencyConstraints()
+            constraints += [RepositoryPackageConstraint(container: specifier, requirement: .unversioned(dependencies))]
```

Comment:
Yep thats right, editing a package doesn't kill its pin. Thats important because editing is a "local" and temporary thing to do where as pins file is supposed to be shared.